prov/usnic/src/fi_ext_usnic.h

```diff
@@ -59,10 +67,40 @@ struct fi_usnic_info_v1 {
 	uint32_t ui_cq_per_vf;
 };
 
+struct fi_usnic_info_v2 {
+	char		ui_devname[USDF_MAX_DEVNAME];
+	char		ui_ifname[IFNAMSIZ];
+	uint8_t		ui_mac_addr[6];
+
+	uint32_t	ui_ipaddr_be;
+	uint32_t	ui_netmask_be;
+	uint32_t	ui_prefixlen;
+	uint32_t	ui_mtu;
+	uint8_t		ui_link_up;
+
+	uint32_t	ui_device_id;
+	char		ui_firmware[64];
+
+	unsigned	ui_num_vf;
+	unsigned	ui_cq_per_vf;
+	unsigned	ui_qp_per_vf;
+	unsigned	ui_intr_per_vf;
+	unsigned	ui_max_cq;
+	unsigned	ui_max_qp;
+
+	uint32_t	ui_link_speed;
+	unsigned	ui_max_cqe;
+	unsigned	ui_max_send_credits;
+	unsigned	ui_max_recv_credits;
+
+	struct usdf_cap	ui_caps[USDF_CAP_MAX];
```

Comment:
How about returning a dynamically allocated array and avoiding this CAP_MAX constant altogether?  The user would be responsible for freeing it, but that's not particularly onerous.

Otherwise, if you stick with a static size, let's bump it up from 5.  We already have 4 or 5 capabilities right now, one/zero spare slot isn't a lot of headroom for the future.  Maybe 10 or 15 instead?